### PR TITLE
chore: Convert local SSL certificate errors into structured logging

### PR DESCRIPTION
**Pull request by izontm**
## Description
This PR makes structured errors around local SSL certificate, including links to documentation.

Original:
image

New:
image

## Related Issues

Related to #15019

### CODE REVIEW

File: packages/gatsby/src/utils/get-ssl-cert.js [hunk 1]
```diff
@@ -15,9 +15,12 @@ module.exports = async ({ name, certFile, keyFile, directory }) => {
   // check that cert file and key file are both true or both false, if they are both
   // false, it defaults to the automatic ssl
   if (certFile ? !keyFile : keyFile) {
-    report.panic(
-      `for custom ssl --https, --cert-file, and --key-file must be used together`
-    )
+    report.panic({
+      id: `11521`,
+      context: {
```
Comment (wardpeet) [inline packages/gatsby/src/utils/get-ssl-cert.js]: Is this bit necesary? maybe empty object or just remove it as we're not using it in the message.

Wdyt?

**Comment by ANDIKKY** [comment]
This would allow for an instance of gatsby develop to be served from behind a proxy server while allowing the socket connection to go directly back to the instance instead of through the proxy...

File: packages/gatsby/src/utils/get-ssl-cert.js (hunk 1 above)
Comment (izontm) [reply packages/gatsby/src/utils/get-ssl-cert.js]: I had tried without passing context, but apparently the message is needed for the gatsby-telemetry package.  I couldn't solve it any other way.  If you have any idea how to overcome this, I can try to solve it.  Thanks.

Comment (pieh) [reply packages/gatsby/src/utils/get-ssl-cert.js]: Let's use empty `context: {}` - this should work ok. But this is separate issue, and we should fix this in telemetry package to not crash on undefined context

**Pull request merged**